### PR TITLE
Fixes to have GPMD as a lib

### PR DESCRIPTION
**Pull request by cnegre**
Two mayor issues corrected:

- Files that were not closed after usage 
- The electronic structure type of the System type was not deallocated.

- - -
This change is [Reviewable](https://reviewable.io/reviews/lanl/qmd-progress/291)

### CODE REVIEW

File: src/prg_system_mod.F90 [hunk 1]
```diff
@@ -2658,26 +2658,30 @@ subroutine prg_graph2bml(graph,bml_type,g_bml)
...
+        row(graph(ii,i)) = 1.0_dp
         !       call bml_set_element_new(g_bml,i,graph(ii,i),1.0)
         !       call bml_set_element_new(g_bml,graph(ii,i),i,1.0)
         ii = ii+1
       enddo
       call bml_set_row(g_bml,i,row,0.5_dp)
     enddo
     !$omp end parallel do
-
+!    stop
```
Comment (nicolasbock) [inline src/prg_system_mod.F90:2677]: Why the commented out stop here? Is that a leftover debug statement?

Comment (cnegre) [reply src/prg_system_mod.F90]: Should be fixed now

File: src/prg_system_mod.F90 [hunk 2]
```diff
@@ -2658,26 +2658,30 @@ subroutine prg_graph2bml(graph,bml_type,g_bml)
     allocate(row(nats))
 
     mdim = bml_get_m(g_bml)
-
+    write(*,*)"mdim",mdim
```
Comment (mewall) [inline src/prg_system_mod.F90:2661]: Is this needed?

File: src/prg_system_mod.F90 [hunk 3]
```diff
@@ -2658,26 +2658,30 @@ subroutine prg_graph2bml(graph,bml_type,g_bml)
...
       enddo
       call bml_set_row(g_bml,i,row,0.5_dp)
     enddo
     !$omp end parallel do
-
+    
     deallocate(graph)
     deallocate(row)
 
+   call bml_print_matrix("gcov",g_bml,0,4,0,4)
```
Comment (mewall) [inline src/prg_system_mod.F90:2681]: Are lines 2681-2683 needed?

**Review by mewall — CHANGES_REQUESTED** [review]
I confirmed that gpmdcov runs on a test system with these changes and shows good energy conservation

It looks like there are a couple of debug outputs that can be removed

File: src/prg_system_mod.F90 (hunk 3 above)
Comment (cnegre) [reply src/prg_system_mod.F90]: Should be fixed now

**Review by nicolasbock — APPROVED** [review]

**Pull request merged**